### PR TITLE
Changed maven insert location for expo plugin.

### LINKED ISSUE #2415: [Bug]: Expo build fails when expo-camera installed
### Mapbox Implementation

Maplibre

### Mapbox Version

default

### Platform

Android

### `@rnmapbox/maps` version

10.0.0-beta.57

### Standalone component to reproduce

N/A

### Observed behavior and steps to reproduce

1. [create new Expo app](https://docs.expo.dev/get-started/create-a-new-app/)
2. [add rnmapbox and configure plugin](https://github.com/rnmapbox/maps/blob/main/plugin/install.md)
3. [add Expo Camera](https://docs.expo.dev/versions/latest/sdk/camera/) (no component required, dependency is enough)
4. [configure EAS Build](https://docs.expo.dev/build/setup/)
5. Start build for Android: `npx eas build --platform android`

Build fails at `Run gradlew`:

```
[stderr] FAILURE: Build failed with an exception.
[stderr] * Where:
[stderr] Build file '/home/expo/workingdir/build/android/build.gradle' line: 40
[stderr] * What went wrong:
[stderr] A problem occurred evaluating root project 'mapbox-build-test'.
[stderr] > Could not find method maven() for arguments [build_a4pmw2504rhdm3lg68kv7pbd1$_run_closure1$_closure3@4911148e] on root project 'mapbox-build-test' of type org.gradle.api.Project.
```

### Expected behavior

Expected build to continue to succeed.

### Notes / preliminary analysis

I assume this is related to #2393 and further work needs to be done on the build.gradle patching?

### Additional links and references

Demo repo at https://github.com/tybot/rnmapbox-with-camera-test

### PR DESCRIPTION
**Pull request by michaek**
## Change maven injection strategy for Expo plugin

Fixes #2415

## Checklist

- [ ] I have tested this on a device/simulator for each compatible OS
- [ ] I updated the documentation with running `yarn generate` in the root folder
- [ ] I updated the typings files - if js interface was changed (`index.d.ts`)
- [ ] I added/ updated a sample - if a new feature was implemented (`/example`)

This PR is a follow-up to #2393, addressing the continued incidence of the underlying problem (see #2415). Instead of trying to find the "right" line to inject the maven configuration, this follows the pattern used in expo-camera to create a new `allprojects` block at the bottom of build.gradle, to avoid the complexity of finding the right nesting level by line.

An additional test is added to show that this handles the failure case reported in #2415, where the presence of expo-camera configuration caused the maven configuration to be injected in the wrong location.

### CODE REVIEW

**Comment by michaek** [comment]
I didn't make any changes to the pods configuration, so I wouldn't expect this change to cause the CI failure above.

**Comment by mfazekas** [comment]
> I didn't make any changes to the pods configuration, so I wouldn't expect this change to cause the CI failure above.

Yep the CI issue is network related, and unrelated to the changes

**Comment by mfazekas** [comment]
@KiwiKilian Can you please review this change? Thanks much!

File: plugin/build/withMapbox.js [hunk 1]
```diff
@@ -208,67 +208,54 @@ const addLibCppFilter = (appBuildGradle) => {
...
-    Should look like this:
-    
-    allprojects {
-      configurations.all {
-          resolutionStrategy {
-              force \\"com.facebook.react:react-native:\\" + REACT_NATIVE_VERSION
-          }
+// Because we need the package to be added AFTER the React and Google maven packages, we create a new allprojects.
+// It's ok to have multiple allprojects.repositories, so we create a new one since it's cheaper than tokenizing
+// the existing block to find the correct place to insert our camera maven.
```
Comment (KiwiKilian) [inline plugin/build/withMapbox.js:213]:
```suggestion
// the existing block to find the correct place to insert our mapbox maven.
```

Comment (michaek) [reply plugin/build/withMapbox.js:213]: Looks like you caught a copy/paste error I didn't!

**Comment by KiwiKilian** [comment]
I'm not much of an expert in the Gradle stuff. But I tested, build's successfully with and without expo-camera using this fix. As it's mainly taken from expo-camera, I would say it's good to go.

**Comment by michaek** [comment]
I should update the comment text in the .ts source too.

**Comment by KiwiKilian** [comment]
Sorry, should have suggested on the correct file, my fault 🙈.

**Review by KiwiKilian — APPROVED** [review]

**Pull request merged**